pkg/workflows/steps/amazon/create_keypair.go

```
@@ -2,94 +2,68 @@ package amazon
 
 import (
 	"context"
+	"fmt"
 	"io"
-	"strings"
+
+	"github.com/pkg/errors"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/ec2"
-	"github.com/pkg/errors"
-	"github.com/sirupsen/logrus"
-	"github.com/supergiant/supergiant/pkg/clouds"
-	"github.com/supergiant/supergiant/pkg/model"
 	"github.com/supergiant/supergiant/pkg/util"
 	"github.com/supergiant/supergiant/pkg/workflows/steps"
 )
 
+const StepName = "awskeypairstep"
+
 //KeyPairStep represents creation of keypair in aws
 //since there is hard cap on keypairs per account supergiant will create one per clster
 type KeyPairStep struct {
-	accounts accountWrapper
 }
 
-func NewKeyPairStep(wrapper accountWrapper) *KeyPairStep {
-	return &KeyPairStep{
-		accounts: wrapper,
-	}
+func NewKeyPairStep() *KeyPairStep {
+	return &KeyPairStep{}
 }
 
 //InitCreateKeyPair add the step to the registry
-func InitCreateKeyPair(wrapper accountWrapper) {
-	steps.RegisterStep(StepNameKeyPair, NewKeyPairStep(wrapper))
-}
-
-const StepNameKeyPair = "keypairstep"
-
-type accountWrapper interface {
-	Get(context.Context, string) (*model.CloudAccount, error)
-	Update(context.Context, *model.CloudAccount) error
+func InitCreateKeyPair() {
+	steps.RegisterStep(StepName, NewKeyPairStep())
 }
 
 func (s *KeyPairStep) Run(ctx context.Context, w io.Writer, cfg *steps.Config) error {
 	log := util.GetLogger(w)
 	log.Infof("[%s] - started!", s.Name())
 
-	account, err := s.accounts.Get(ctx, cfg.CloudAccountName)
-	if err != nil {
-		return errors.Wrap(err, "aws: no cloud account found")
-	}
-
 	sdk, err := GetSDK(cfg.AWSConfig)
 	if err != nil {
 		return errors.New("aws: authorization")
 	}
 
+	// Create key for user
+	userKeyPairName := fmt.Sprintf("%s-user", cfg.AWSConfig.KeyPairName)
+
+	req := &ec2.ImportKeyPairInput{
+		KeyName: &userKeyPairName,
+		PublicKeyMaterial: []byte(cfg.SshConfig.PublicKey),
+	}
+
+	output, _ := sdk.EC2.ImportKeyPairWithContext(ctx, req)
```

Why do you ignore this error?

---

Because key can be already registred, but if it is not import it.

---

```
// Returns awserr.Error for service API and SDK errors. Use runtime type assertions
// with awserr.Error's Code and Message methods to get detailed information about
// the error.
```
Have you reviewed it? Probably, it could have a particular error code for this case.